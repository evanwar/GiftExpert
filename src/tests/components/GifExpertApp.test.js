import { shallow } from "enzyme";
import GifExpertApp from "../../GifExpertApp";

describe('GifExpertApp', () => {
    test('SnapShot', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });

    test('should be show a list of categories', () => {

        const categories=['Ready Player One','Zelda'];
        const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    });
})
