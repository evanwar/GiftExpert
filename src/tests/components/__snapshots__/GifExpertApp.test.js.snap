// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp SnapShot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpertApp should be show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Ready Player One"
      key="Ready Player One"
    />
    <GifGrid
      category="Zelda"
      key="Zelda"
    />
  </ol>
</Fragment>
`;
